lib/PuppeteerSharp/DOMWorld.cs

```diff
@@ -48,7 +48,7 @@ public DOMWorld(CDPSession client, FrameManager frameManager, Frame frame, Timeo
             _logger = _client.Connection.LoggerFactory.CreateLogger<DOMWorld>();
         }
 
-        internal ICollection<WaitTask> WaitTasks { get; set; }
+        internal ConcurrentSet<WaitTask> WaitTasks { get; set; }
```

Comment:
Changing the type of this internal property was necessary to get access to `IsEmpty` at line 146.